kt/godot-runtime/src/main/kotlin/godot/core/Properties.kt

```diff
@@ -62,10 +68,27 @@ class KtEnumProperty<T : KtObject, P : Any>(
     }
 
     override fun callSet(instance: T) {
-        val argsSize = TransferContext.buffer.int
-        require(argsSize == 1) { "Setter should be called with only one argument." }
-        val arg = TransferContext.readSingleArgument(VariantType.JVM_INT)
-        TransferContext.buffer.rewind()
-        kProperty.set(instance, setValueConverter(arg as Int))
+        val arg = extractSetterArgument<Int>()
+        kProperty.set(instance, setValueConverter(arg))
+    }
+}
+
+class KtArrayProperty<T: KtObject, P: Any?>(

```

Comment:
This is not used atm. is it needed?

Reply:
Was that to store the type of the Array? 
If that's the case, it's not necessary anymore, we already have the type information stored in VariantArray

Reply:
was to register array, but is deprecated, I removed it.